### PR TITLE
feat!: replace HADMap with Lanelet

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

Replace HADMap of autoware_auto_planning_msg with Lanelet of autoware_planning_msgs.
LaneLetRoute has uuid and I want to use it in planning modules. Please refer to https://github.com/autowarefoundation/autoware/discussions/2983 for details.

## Related links

- depends autoware_common
  - https://github.com/autowarefoundation/autoware_common/pull/133 
- autoware_launch
  - https://github.com/tier4/autoware_launch/pull/571  
- port PR of auto_msgs to autoware_msgs
  - https://github.com/autowarefoundation/autoware_msgs/pull/33
  - Rename to Lanelet
    -  https://github.com/autowarefoundation/autoware_msgs/pull/41  
  - Add uuid to Lanelet
    - https://github.com/autowarefoundation/autoware_msgs/pull/42
    
## Tests performed

psim

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kosuke55** [comment]
this PR depends on https://github.com/autowarefoundation/autoware_common/pull/133, so merge it first.

**Comment by yukkysaito** [comment]
@kosuke55 Can you resolve some conflicts?

**Comment by isamu-takagi** [comment]
@kosuke55 The ad_service_state_monitor package is replaced with AD API. So just remove it to resolve the conflict.

**Comment by isamu-takagi** [comment]
@kosuke55 Sorry, config files need to be updated.
* https://github.com/autowarefoundation/autoware.universe/blob/main/system/component_state_monitor/config/topics.yaml
* https://github.com/autowarefoundation/autoware.universe/blob/main/launch/tier4_system_launch/config/component_state_monitor/topics.yaml

**Comment by kosuke55** [comment]
@isamu-takagi @yukkysaito 
thanks, I resolved conflicts and updated the yaml.

**Comment by kosuke55** [comment]
oh, sorry build failed I will check it

**Review by isamu-takagi — APPROVED** [review]
LGTM. Checked it in the simple planning simulator.

**Review by yukkysaito — APPROVED** [review]
LGTM

File: perception/traffic_light_map_based_detector/README.md [hunk 1]
```diff
@@ -13,11 +13,11 @@ If the node receives no route information, it looks at a radius of 200 meters an
...
-| Name                 | Type                                     | Description             |
-| -------------------- | ---------------------------------------- | ----------------------- |
-| `~input/vector_map`  | autoware_auto_mapping_msgs::HADMapBin    | vector map              |
-| `~input/camera_info` | sensor_msgs::CameraInfo                  | target camera parameter |
-| `~input/route`       | autoware_auto_planning_msgs::HADMapRoute | optional: route         |
+| Name                 | Type                                      | Description             |
+| -------------------- | ----------------------------------------- | ----------------------- |
+| `~input/vector_map`  | autoware_auto_mapping_msgs::HADMapBin     | vector map              |
+| `~input/camera_info` | sensor_msgs::CameraInfo                   | target camera parameter |
+| `~input/route`       | autoware_auto_planning_msgs::LaneletRoute | optional: route         |
```
Comment (wep21) [inline perception/traffic_light_map_based_detector/README.md:20]:
```suggestion
| `~input/route`       | autoware_planning_msgs::LaneletRoute | optional: route         |
```

File: planning/behavior_path_planner/README.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ The following modules are currently supported:
 
 ### input
 
-- /planning/mission_planning/route [`autoware_auto_planning_msgs/HADMapRoute`] : Current route from start to goal.
+- /planning/mission_planning/route [`autoware_auto_planning_msgs/LaneletRoute`] : Current route from start to goal.
```
Comment (wep21) [inline planning/behavior_path_planner/README.md:46]:
```suggestion
- /planning/mission_planning/route [`autoware_planning_msgs/LaneletRoute`] : Current route from start to goal.
```

File: planning/mission_planner/README.md [hunk 3]
```diff
@@ -38,23 +38,23 @@ In current Autoware.universe, only Lanelet2 map format is supported.
...
 
-| Name                            | Type                                    | Description            |
-| ------------------------------- | --------------------------------------- | ---------------------- |
-| `/planning/routing/route_state` | autoware_adapi_v1_msgs::msg::RouteState | route state            |
-| `/planning/routing/route`       | autoware_auto_planning_msgs/HADMapRoute | route                  |
-| `debug/route_marker`            | visualization_msgs::msg::MarkerArray    | route marker for debug |
+| Name                            | Type                                     | Description            |
+| ------------------------------- | ---------------------------------------- | ---------------------- |
+| `/planning/routing/route_state` | autoware_adapi_v1_msgs::msg::RouteState  | route state            |
+| `/planning/routing/route`       | autoware_auto_planning_msgs/LaneletRoute | route                  |
```
Comment (wep21) [inline planning/mission_planner/README.md:44]:
```suggestion
| `/planning/routing/route`       | autoware_planning_msgs/LaneletRoute | route                  |
```

File: planning/scenario_selector/README.md [hunk 4]
```diff
@@ -6,14 +6,14 @@
...
-| `~input/odometry`                | nav_msgs::Odometry                       | for checking whether vehicle is stopped               |
-| `is_parking_completed`           | bool (implemented as rosparam)           | whether all split trajectory of Parking are published |
+| Name                             | Type                                      | Description                                           |
+| -------------------------------- | ----------------------------------------- | ----------------------------------------------------- |
+| `~input/lane_driving/trajectory` | autoware_auto_planning_msgs::Trajectory   | trajectory of LaneDriving scenario                    |
+| `~input/parking/trajectory`      | autoware_auto_planning_msgs::Trajectory   | trajectory of Parking scenario                        |
+| `~input/lanelet_map`             | autoware_auto_mapping_msgs::HADMapBin     |                                                       |
+| `~input/route`                   | autoware_auto_planning_msgs::LaneletRoute | route and goal pose                                   |
```
Comment (wep21) [inline planning/scenario_selector/README.md:14]:
```suggestion
| `~input/route`                   | autoware_planning_msgs::LaneletRoute | route and goal pose                                   |
```

**Pull request merged**